### PR TITLE
RSDK-3306 - arm tests

### PR DESCRIPTION
**Pull request by clintpurser**
- fakeArm
- Tests (FakeArm, Service, Client)
- fix `extra` in arm client
- implement doCommand in arm client

### CODE REVIEW

**Review by njooma — APPROVED** [review]

File: test/unit_test/components/arm_test.dart [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+import 'package:viam_sdk/src/gen/component/arm/v1/arm.pbgrpc.dart';
+import 'package:viam_sdk/src/resource/manager.dart';
+import 'package:viam_sdk/src/utils.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+
+import '../../test_utils.dart';
+
+class FakeArm extends Arm {
+  bool isStopped = true;
+  JointPositions armJoinPositions = JointPositions(values: [0, 0, 0]);
```
Comment (purplenicole730) [inline test/unit_test/components/arm_test.dart:13]: small thing, but I was curious why it's named `JoinPositions`?

File: test/unit_test/components/arm_test.dart [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+      test('extra', () async {
+        expect(arm.extra, null);
+
+        final client = ArmServiceClient(channel);
+        await client.stop(StopRequest(name: name, extra: {'foo': 'bar'}.toStruct()));
+        expect(arm.extra, {'foo': 'bar'});
+      });
+    });
+    group('Arm Client Tests', () {
```
Comment (purplenicole730) [inline test/unit_test/components/arm_test.dart:228]: nit: Is there supposed to be a line before this?

**Review by purplenicole730 — APPROVED** [review]
LGTM! (with my nonexistent dart language knowledge)

File: test/unit_test/components/arm_test.dart (hunk 1 above)
Comment (clintpurser) [reply test/unit_test/components/arm_test.dart:13]: that is a typo, thank you for finding that :)

File: test/unit_test/components/arm_test.dart (hunk 2 above)
Comment (clintpurser) [reply test/unit_test/components/arm_test.dart:228]: for consistency yes you're right, thank you!

**Pull request merged**